lib/fhir_models/bootstrap/model.rb

```diff
@@ -300,6 +300,26 @@ def check_binding_uri(uri, value)
       valid
     end
 
+    def each_element(path = nil, &block)
+      self.class::METADATA.each do |element_name, metadata|
+        local_name = metadata.fetch :local_name, element_name
+        values = [instance_variable_get("@#{local_name}")].flatten.compact
+        next if values.empty?
+
+        values.each_with_index do |value, i|
+          child_path =
+            if path.nil?
+              element_name
+            else
+              "#{path}.#{element_name}"
+            end
+          child_path += "[#{i}]" if metadata['max'] > 1
+          yield value, metadata, child_path
+          value.each_element child_path, &block unless FHIR::PRIMITIVES.include? metadata['type']
+        end
+      end
```

Comment:
@arscan mentioned that the return value is a little weird and I agree (it threw me off when I was testing)

A typical `#each` would return the object itself. Any reason not to do that here?

```suggestion
      end
      self
```